sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java

```diff
@@ -146,17 +160,38 @@ void startPartitionPump(PartitionOwnership claimedOwnership, Checkpoint checkpoi
                 .createConsumer(claimedOwnership.getConsumerGroup(), EventHubClientBuilder.DEFAULT_PREFETCH_COUNT);
 
             partitionPumps.put(claimedOwnership.getPartitionId(), eventHubConsumer);
-            eventHubConsumer
-                .receiveFromPartition(claimedOwnership.getPartitionId(), startFromEventPosition, receiveOptions)
-                .subscribe(partitionEvent -> processEvent(partitionContext, partitionProcessor, eventHubConsumer,
-                    partitionEvent),
-                    /* EventHubConsumer receive() returned an error */
-                    ex -> handleError(claimedOwnership, eventHubConsumer, partitionProcessor, ex, partitionContext),
-                    () -> {
-                        partitionProcessor.close(new CloseContext(partitionContext,
-                            CloseReason.EVENT_PROCESSOR_SHUTDOWN));
-                        cleanup(claimedOwnership, eventHubConsumer);
-                    });
+            if (maxWaitTime != null) {
```

Comment:
What happens if they set Duration.ZERO? Would we try the window timeout approach?

Reply:
Added validation to check that duration is not zero and also switched to window timeout as that seems like the more suited operator here.